src/client/Microsoft.Identity.Client/ManagedIdentity/V2/ImdsV2ManagedIdentitySource.cs

```diff
@@ -309,5 +353,92 @@ private static string ImdsV2QueryParamsHelper(RequestContext requestContext)
 
             return queryParams;
         }
+
+        private async Task<string> GetAttestationJwtAsync(
+            string clientId, 
+            Uri attestationEndpoint, 
+            ManagedIdentityKeyInfo keyInfo, 
+            CancellationToken cancellationToken)
+        {
+            // Provider is a local dependency; missing provider is a client error (not service)
+            var provider = _requestContext.AttestationTokenProvider;
+
+            // KeyGuard requires RSACng on Windows
+            if (keyInfo.Type == ManagedIdentityKeyType.KeyGuard &&
+                keyInfo.Key is not System.Security.Cryptography.RSACng rsaCng)
+            {
+                throw new MsalClientException(
+                    "keyguard_requires_cng",
+                    "[ImdsV2] KeyGuard attestation currently supports only RSA CNG keys on Windows.");
+            }
+
+            var input = new AttestationTokenInput
+            {
+                ClientId = clientId,
+                AttestationEndpoint = attestationEndpoint,
+                KeyHandle = (keyInfo.Key as System.Security.Cryptography.RSACng)?.Key.Handle
+            };
+
+            var response = await provider(input, cancellationToken).ConfigureAwait(false);
+            if (response == null || string.IsNullOrWhiteSpace(response.AttestationToken))
+            {
+                throw new MsalClientException(
+                    "attestation_empty",
+                    "[ImdsV2] Attestation provider returned an empty token.");
+            }
+
+            return response.AttestationToken;
+        }
+
+        /// <summary>
+        /// Temporarily normalize attestation endpoint values to a full https:// URI.
+        /// IMDS team will eventually return a full URI. 
```

Comment:
Will we be eventually deleting this function? If so, can you add "TODO" in this code?

Reply:
yes, added